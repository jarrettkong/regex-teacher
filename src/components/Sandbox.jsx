import React, { Component } from 'react'
import SandboxForm from './SandboxForm.jsx'
import SandboxResults from './SandboxResults.jsx'
import Nav from './Nav.jsx'
import './Sandbox.scss'

class Sandbox extends Component {

  constructor(props) {
    super(props)
    this.state = {
      pattern: '',
      flags: '',
      inputValue: '',
      matches:[]
    }
  }

  _handleForm = (pattern, flags) => {
    this.setState({ pattern, flags }, () => {
      const matches = this.getMatches();
      if(!matches){
        return;
      }
      this.setState({ matches })
    })
  }

  _handleTextArea = e => {
    this.setState({ inputValue: e.target.value }, () => {
      const matches = this.getMatches();
      if(!matches) {
        return;
      }
      this.setState({ matches })
    })
  }

  getMatches = () => {
    const { pattern, flags, inputValue } = this.state;
    try {
      const regex = new RegExp(pattern, flags);
      const matches = inputValue.match(regex);
      return matches;
    } catch(err) {
      return
    }
  }

  render() {
    return (
      <main className="Sandbox">
        <Nav />
        <h1 className="Sandbox-header">Sandbox</h1>
        <p>Test your regular expressions with a custom input</p>
        <section className="Sandbox-area">
          <SandboxForm handleTextArea={this._handleTextArea} handleForm={this._handleForm}/>
          <SandboxResults matches={this.state.matches}/>
        </section>
      </main>
    )
  }
}

export default Sandbox
