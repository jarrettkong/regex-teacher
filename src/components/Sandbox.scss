.Sandbox {
  height: 100vh;
  max-width: 1280px;
  margin: auto;
  color: white;
  padding: 50px 25px;
}

.Sandbox-header {
  margin: 0 0 25px 0;
  font-size: 50px;
  line-height: 100%;
}

.Sandbox-area {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 30px;
}