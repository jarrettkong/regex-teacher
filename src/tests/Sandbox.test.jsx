import React from 'react';
import Sandbox from '../components/Sandbox.jsx'
import { shallow } from 'enzyme';

describe('SandboxForm', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Sandbox />
    )
  });

  it('should render the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper default state', () => {
    expect(wrapper.state()).toEqual({
      pattern: '',
      flags: '',
      inputValue: '',
      matches: []
    });
  })

});

