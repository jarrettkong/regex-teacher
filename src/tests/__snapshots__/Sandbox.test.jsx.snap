// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SandboxForm should render the snapshot with all the data passed in 1`] = `
<main
  className="Sandbox"
>
  <Nav />
  <h1
    className="Sandbox-header"
  >
    Sandbox
  </h1>
  <p>
    Test your regular expressions with a custom input
  </p>
  <section
    className="Sandbox-area"
  >
    <SandboxForm
      handleForm={[Function]}
      handleTextArea={[Function]}
    />
    <SandboxResults
      matches={Array []}
    />
  </section>
</main>
`;
